cassandra-mesos-scheduler/src/main/java/io/mesosphere/mesos/frameworks/cassandra/scheduler/SeedManager.java

```diff
@@ -0,0 +1,110 @@
+package io.mesosphere.mesos.frameworks.cassandra.scheduler;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Joiner;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import static com.google.common.collect.Lists.newArrayList;
+import static io.mesosphere.mesos.frameworks.cassandra.CassandraFrameworkProtos.CassandraFrameworkConfiguration;
+import static io.mesosphere.mesos.frameworks.cassandra.CassandraFrameworkProtos.ExternalDc;
+
+public class SeedManager {
+    public static final long FETCH_PERIOD = TimeUnit.MINUTES.toMillis(1);
+    private static final Logger LOGGER = LoggerFactory.getLogger(SeedManager.class);
+
+    @NotNull
+    private final PersistedCassandraFrameworkConfiguration configuration;
+
+    public SeedManager(@NotNull PersistedCassandraFrameworkConfiguration configuration) {
+        this.configuration = configuration;
+    }
+
+    @NotNull
+    public List<String> getSeeds() {
+        final List<String> seeds = newArrayList();
+
+        for (final ExternalDc dc : configuration.get().getExternalDcsList()) {
+            seeds.addAll(dc.getSeedsList());
+        }
+
+        return seeds;
+    }
+
+    public void syncSeeds(long now) {
+        for (ExternalDc dc : configuration.get().getExternalDcsList()) {
+            fetchSeedsIfRequired(dc, now);
+        }
+    }
+
+    private void fetchSeedsIfRequired(ExternalDc dc, long now) {
+        long lastTime = dc.getSeedFetchTime();
+        if (now - lastTime < FETCH_PERIOD) return;
+
+        String url = dc.getUrl();
+        if (url.endsWith("/")) url = url.substring(0, url.length() - 1);
+        url += "/node/seed/all";
+
+        JsonNode node = fetchJson(url);
+        if (node == null) return;
+
+        JsonNode seedsArr = node.get("seeds");
+        List<String> seeds = newArrayList();
+        for (JsonNode ip : seedsArr) {
+            seeds.add(ip.asText());
+        }
+
+        LOGGER.info("Got seeds for dc \"" + dc.getName() + "\": " + Joiner.on(", ").join(seeds));
+        updateDcSeeds(dc, seeds, now);
+    }
+
+    protected JsonNode fetchJson(String url) {
+        HttpURLConnection c = null;
+        try {
+            c = (HttpURLConnection)new URL(url).openConnection();
+            ObjectMapper mapper = new ObjectMapper();
```

Comment:
Note sure about this. Avg creation of ObjectMapper takes ~ 0.031 ms on my laptop. Please see the output:
```
import com.fasterxml.jackson.databind.ObjectMapper;

public class T {
    public static void main(String[] args) {
        long codeRef = 0, start = System.currentTimeMillis();
        int creations = 10000;

        for (int i = 0; i < creations; i++) {
            ObjectMapper mapper = new ObjectMapper();
            codeRef += mapper.hashCode();
        }

        long time = System.currentTimeMillis() - start;
        double avg = Math.floor(time / (double)creations * 1000) / 1000;

        System.out.println(codeRef);
        System.out.println(creations + " creations of ObjectMapper took " + time + " ms, avg took " + avg + " ms");
    }
}
```
Output:
```
10700061484741
10000 creations of ObjectMapper took 310 ms, avg took 0.031 ms
```

IMHO, no need to hold an instance in this case.